wave_vision/include/wave/vision/utils.hpp

```diff
@@ -43,20 +43,54 @@ Vec2 focal_length(double hfov,
  */
 void projection_matrix(const Mat3 &K, const Mat3 &R, const Vec3 &t, MatX &P);
 
-/** Convert a single cv::KeyPoint to a Vec2
+/** Convert a single cv::KeyPoint to Vec2
  *
  * @param keypoint input keypoint
  * @param vec_keypoints converted keypoint
  */
-void convertKeypoint(const cv::KeyPoint &keypoint, Vec2 &vec_keypoint);
+Vec2 convertKeypoint(const cv::KeyPoint &keypoint);
 
-/** Convert a vector of keypoints to a vector of Vec2
+/** Convert a single cv::Point2f to Vec2
+ *
+ * @param keypoint input keypoint
+ * @param vec_keypoints converted keypoint
+ */
+Vec2 convertKeypoint(const cv::Point2f &keypoint);
+
+/** Convert a single Vec2 to cv::Point2f
+ *
+ * @param keypoint input keypoint
+ * @param vec_keypoints converted keypoint
+ */
+cv::Point2f convertKeypoint(const Vec2 &keypoint);
+
+/** Convert a vector of cv::KeyPoint to a vector of Vec2
  *
  * @param keypoints input keypoints
  * @param vec_keypoints converted keypoints
  */
-void convertKeypoints(const std::vector<cv::KeyPoint> &keypoints,
-                      std::vector<Vec2> &vec_keypoints);
+std::vector<Vec2> convertKeypoints(const std::vector<cv::KeyPoint> &keypoints);
```

Comment:
I'm using one of them I think. I felt it was just useful to have for future use - these are the three main types a user may encounter in terms of keypoints.